CONTRIBUTING.md

```diff
@@ -0,0 +1,139 @@
+
+The Activity Log page is using the MVC architecture and [web components](https://github.com/mozilla/extension-activity-monitor/tree/master/src/lib/web-component) for table view and filtering options. ES6 classes/modules are being used to encapsulate the different functionalities of the extension. Unit test is done with [JEST Framework](https://jestjs.io/) in this project.
+
+### Data Flow
+
+The extension recevies logs in the form of `object` from activityLog API. The activityLog API schema can been found [here](https://searchfox.org/mozilla-central/source/toolkit/components/extensions/schemas/activity_log.json). It has only one event called [`onExtensionActivity`](https://searchfox.org/mozilla-central/source/toolkit/components/extensions/schemas/activity_log.json#20), which gets triggered everytime an activity is found from any monitored extensions and return a [log object](https://searchfox.org/mozilla-central/source/toolkit/components/extensions/schemas/activity_log.json#24-76). We store these log objects inside the [`logs` array in background](https://github.com/mozilla/extension-activity-monitor/blob/master/src/lib/ext-monitor.js#L7). We also stored the monitored extension in a [`Map` in Background](https://github.com/mozilla/extension-activity-monitor/blob/master/src/lib/ext-monitor.js#L9). When we open the Activity Log page from popup, the page is being rendered with existing logs collected in the background and receives the future logs in real-time via [`runtime.onMessage` event](https://github.com/mozilla/extension-activity-monitor/blob/master/src/lib/ext-activitylog.js#L235-L247) while it is opened. In Activity Log page, the logs are being rendered in table view with the help of [`log-view` web component](https://github.com/mozilla/extension-activity-monitor/blob/master/src/lib/web-component/log-view/log-view-element.js).
+
+### Project Directory Structure
+
+- [_`src`_](https://github.com/mozilla/extension-activity-monitor/tree/master/src) - Inside _src_ the extension's code can be found (manifest.json, background.js etc.)
+  - [_`activitylog`_](https://github.com/mozilla/extension-activity-monitor/tree/master/src/activitylog) - Contains activity log page (extension page).
+  - [_`devtools`_](https://github.com/mozilla/extension-activity-monitor/tree/master/src/devtools) - Contains the invisible devtools page that creates a devtools panel which loads the activitylog.html page and filter logs with the current tab id.
+  - [_`popup`_](https://github.com/mozilla/extension-activity-monitor/tree/master/src/popup) - Contains popup (browserAction popup).
+  - [_`icons`_](https://github.com/mozilla/extension-activity-monitor/tree/master/src/icons) - Contains icons used in the extension.
+  - [_`lib`_](https://github.com/mozilla/extension-activity-monitor/tree/master/src/lib) - Contains most of the project's code. Here is a brief description of each of the files:
+    - `ext-monitor.js` - It runs in the background.
```

Comment:
These descriptions on their own aren't particularly useful. As mentioned in the top-level review comment, an overview of components (potentially with references to files) is useful.

Just a list of directories and file names is not very useful. The list doesn't show much information and can become out of date.

Since the directory names are self-describing and there isn't that much info here, I recommend to remove this whole section once you've got the overview of components and the extra information that I requested in the top-level comment.